api/commands/run.ts

```diff
@@ -109,12 +109,12 @@ export class Runner {
       )
       .map(fn => fn.name);
 
-    pendingNodes.forEach(node => {
+    await Promise.all(pendingNodes.map(async (node) => {
       const finishedDeps = node.dependencies.filter(d => allFinishedDeps.indexOf(d) >= 0);
       const successfulDeps = node.dependencies.filter(d => allSuccessfulDeps.indexOf(d) >= 0);
       if (!this.cancelled && successfulDeps.length == node.dependencies.length) {
         // All required deps are completed, start this node.
-        this.executeNode(node);
+        await this.executeNode(node);
```

Comment:
i'm a bit concerned about this change; i think it reduces the parallelism of Node execution - i think this means that each iteration of loop() waits for all nodes whose execution was started to finish before starting another iteration, which (i think?) means that fewer nodes will run in parallel and overall execution will take longer.

what we want is two things:
1) as much parallelism in *execution* as possible
2) a happens-before relationship on the onChange handlers (or at least, for resultPromise to not resolve until all change handlers have resolved).

one way we could achieve this is by doing a (fairly major, so we should consider it carefully) refactor of this file to invert the dependency graph: once a node completes successfully, call triggerChange(), and then execute all of its children (and so on). that way, all nodes are running that can be running, at all times, and we enforce that triggerChange() always completes before resultPromise does.

i'm not actually sure I see any obvious alternatives to this though...